### DISCUSSION TITLE
set php memory limit

### DISCUSSION THREAD

**Discussion by jquiros2** [nextcloud/all-in-one#1422] [Questions] [answered]
Hi!  I'm getting some memory limit errors and would like to change the memory_limit without losing data.

What is the proper way?  I found "-e NEXTCLOUD_MEMORY_LIMIT=1024" in the docs about this, but if I re-run the initial startup script, it looks like it will recreate it as it says it already exists and needs a different name.

I also found how to do it in the container through /usr/local/etc/php/conf.d/nextcloud.ini but that is not persistent.

Any basic docker tips about this?

Thanks!

**Comment by szaimen** [comment, marked as answer]
Hi,, you would basically need to stop the mastercontainer, remove it and use the same command that you've used to create the mastercontainer initially but add the environmental variable to that command.

**Reply by logiczny** [reply]
That is correct, I did:
```
sudo docker stop nextcloud-aio-mastercontainer
sudo docker rm nextcloud-aio-mastercontainer
sudo docker run --init --sig-proxy=false --name nextcloud-aio-mastercontainer --restart always --publish 8080:8080 --env APACHE_PORT=11000 --env APACHE_IP_BINDING=0.0.0.0 --env NEXTCLOUD_MEMORY_LIMIT=2048M --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest
```
and after few seconds I need to stop all the containers from AIO gui
`https://x.x.x.x:8080/containers`
and start them again and php memory limit is set to 2GB, as requested.
I've checked this when settings memory for face recognizition:
```
sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ face:setup --memory 1G
System memory: 7.8 GB (8326447104B)
Memory assigned to PHP: 2 GB (2147483648B)

Minimum value to assign to image processing.: 682.7 MB (715827882B)
Maximum value to assign to image processing.: 2 GB (2147483648B)

Maximum memory assigned for image processing: 1 GB (1073741824B)
```
